### PR TITLE
build-script: generate options (NFC)

### PR DESCRIPTION
**Pull request by compnerd**
Rather than individually list the options, generate the options which
ensures that options are more uniform and makes it easier to alter them
globally.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @Rostepher @edymtt @xiaobai

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: utils/build-script-impl [hunk 1]
```diff
@@ -115,94 +115,52 @@ KNOWN_SETTINGS=(
...
-    cmark-cmake-options                           ""                "CMake options used for all cmark targets"
     common-cmake-options                          ""                "CMake options used for all targets, including LLVM/Clang"
     extra-cmake-options                           ""                "Extra options to pass to CMake for all targets"
-    foundation-cmake-options                      ""                "CMake options used for all foundation targets"
-    libdispatch-cmake-options                     ""                "CMake options used for all libdispatch targets"
-    libicu-cmake-options                          ""                "CMake options used for all libicu targets"
-    llbuild-cmake-options                         ""                "CMake options used for all llbuild targets"
-    lldb-cmake-options                            ""                "CMake options used for all lldb targets"
-    llvm-cmake-options                            ""                "CMake options used for all llvm targets"
     ninja-cmake-options                           ""                "CMake options used for all ninja targets"
```
Comment (Rostepher) [inline utils/build-script-impl]: Huh, looks like ninja has a duplicate option here. Not that we need to fix it in this PR, just pointing it out.

File: utils/build-script-impl [hunk 2]
```diff
@@ -294,6 +240,17 @@ KNOWN_SETTINGS=(
     skip-local-host-install                       ""                "If we are cross-compiling multiple targets, skip an install pass locally if the hosts match"
 )
 
+components=( cmark foundation libcxx libdispatch libicu llbuild lldb llvm playgroundsupport static-foundation static-libdispatch swift xctest )
```
Comment (Rostepher) [inline utils/build-script-impl]: Can we drop the extra spaces around the parens?

File: utils/build-script-impl (hunk 1 above)
Comment (compnerd) [reply utils/build-script-impl]: Yeah, I realized that too, and will remove that in a follow up.  This is part of the motivation of re-structuring the options - it allows the opportunity to clean this up.

File: utils/build-script-impl (hunk 2 above)
Comment (compnerd) [reply utils/build-script-impl]: Sure.

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

**Review by Rostepher — APPROVED** [review]
This looks good to me. Thanks for simplifying these options!

**Pull request merged**